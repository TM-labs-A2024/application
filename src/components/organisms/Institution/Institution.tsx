import { ArrowBackIcon } from '@chakra-ui/icons'
import {
  IconButton,
  Text,
  Stack,
  Heading,
  Divider,
  Button,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalFooter,
  ModalBody,
  useDisclosure
} from '@chakra-ui/react'
import { ACCESS_DENIED, ACCESS_GRANTED, ACCESS_REMOVAL } from '@constants/index'
import { Institution as InstitutionType } from '@src/types'
import { isIOS, isAndroid, isMobile } from '@utils/index'
import Image from 'next/image'
import { useRouter } from 'next/navigation'
import React, { ReactElement, useCallback } from 'react'
import { Store } from 'react-notifications-component'

function ConfirmationModal({
  isOpen,
  onClose,
  onSubmit,
  method,
  name
}: {
  isOpen: boolean
  onClose: () => void
  onSubmit: () => void
  method: string
  name: string
}) {
  return (
    <Modal isOpen={isOpen} onClose={onClose}>
      <ModalOverlay />
      <ModalContent>
        <ModalBody>
          ¿Seguro que desea {method.toLowerCase()} el acceso a {name}?
        </ModalBody>

        <ModalFooter>
          <Button mr={3} onClick={onClose}>
            Cancelar
          </Button>
          <Button colorScheme="blackAlpha" onClick={onSubmit}>
            {method} acceso
          </Button>
        </ModalFooter>
      </ModalContent>
    </Modal>
  )
}

export default function Institution({
  institution
}: {
  institution: InstitutionType
}): ReactElement {
  // --- Hooks -----------------------------------------------------------------
  const router = useRouter()
  const { isOpen, onOpen, onClose } = useDisclosure()
  const {
    isOpen: isApprovalOpen,
    onOpen: onApprovalOpen,
    onClose: onApprovalClose
  } = useDisclosure()
  const { isOpen: isRemovalOpen, onOpen: onRemovalOpen, onClose: onRemovalClose } = useDisclosure()
  // --- END: Hooks ------------------------------------------------------------

  // --- Data and handlers -----------------------------------------------------
  const onDenial = useCallback(() => {
    Store.addNotification(ACCESS_DENIED(isMobile(window)))
    onClose()
    router.push('/ministerio/solicitudes')
  }, [onClose, router])

  const onApproval = useCallback(() => {
    Store.addNotification(ACCESS_GRANTED(isMobile(window)))
    onApprovalClose()
    router.push('/ministerio/solicitudes')
  }, [onApprovalClose, router])

  const onRemoval = useCallback(() => {
    Store.addNotification(ACCESS_REMOVAL(isMobile(window)))
    onRemovalClose()
    router.push('/ministerio/instituciones')
  }, [onRemovalClose, router])
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div
      className={`mx-auto block h-screen w-screen overflow-hidden px-8 ${isIOS() ? 'pb-64 pt-20' : 'pb-0 pt-10'} ${isAndroid() ?? 'pb-64 pt-8'}`}
      data-testid="institution"
    >
      <div className={`mb-8 flex flex-row justify-between`}>
        <Image alt="logo" src="/static/images/logo-horizontal.png" width={200} height={80} />
      </div>
      <div className="mb-8 flex flex-row items-center justify-start">
        <IconButton
          size="xl"
          className="mr-4"
          aria-label="back"
          variant="link"
          icon={<ArrowBackIcon />}
          onClick={() => {
            router.back()
          }}
        />
        <Text className="font-medium">Perfil de la institución</Text>
      </div>
      <div className="flex h-4/5 flex-col justify-between lg:px-96">
        <div>
          <Stack spacing={1} mb={6}>
            <Heading as="h3" size="md" noOfLines={1}>
              {institution.name}
            </Heading>
            <Text>RIF {institution.credentials}</Text>
          </Stack>
          <Divider orientation="horizontal" />
          <Stack mb={6}>
            <h4 className="text-sm text-gray-600">Correo electrónico</h4>
            <Text className="font-medium">{institution.email}</Text>
          </Stack>
          <Stack mb={6}>
            <h4 className="text-sm text-gray-600">Teléfono</h4>
            <Text className="font-medium">{institution.phone}</Text>
          </Stack>
        </div>
        {institution?.pending && (
          <div className="mb-4 flex w-full flex-row">
            <Button className="mr-2 flex-grow" variant="outline" onClick={onOpen}>
              Rechazar
            </Button>
            <Button className="ml-2 flex-grow" onClick={onApprovalOpen}>
              Aceptar
            </Button>
          </div>
        )}
        {!institution?.pending && (
          <div className="mb-4 flex w-full flex-row">
            <Button className="flex-grow" variant="outline" onClick={onRemovalOpen}>
              Revocar acceso
            </Button>
          </div>
        )}
      </div>
      <ConfirmationModal
        isOpen={isOpen}
        onClose={onClose}
        onSubmit={onDenial}
        method="Denegar"
        name={institution.name}
      />
      <ConfirmationModal
        isOpen={isApprovalOpen}
        onClose={onApprovalClose}
        onSubmit={onApproval}
        method="Conceder"
        name={institution.name}
      />
      <ConfirmationModal
        isOpen={isRemovalOpen}
        onClose={onRemovalClose}
        onSubmit={onRemoval}
        method="Revocar"
        name={institution.name}
      />
    </div>
  )
}
