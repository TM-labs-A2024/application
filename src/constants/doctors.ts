export const doctors = [
  {
    id: 1000000,
    specialities: [1],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 2000000,
    specialities: [2, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 3000000,
    specialities: [3, 5, 7],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 4000000,
    specialities: [4, 1],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 5000000,
    specialities: [5, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 6000000,
    specialities: [6, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 7000000,
    specialities: [7, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 8000000,
    specialities: [8, 9, 5, 1, 3, 4],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 9000000,
    specialities: [9, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 10000000,
    specialities: [10, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 11000000,
    specialities: [11, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 12000000,
    specialities: [12, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 13000000,
    specialities: [13, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 14000000,
    specialities: [14, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 15000000,
    specialities: [15, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 16000000,
    specialities: [16, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 17000000,
    specialities: [17, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 18000000,
    specialities: [18, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 19000000,
    specialities: [19, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 20000000,
    specialities: [20, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 21000000,
    specialities: [21, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 22000000,
    specialities: [22, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 23000000,
    specialities: [23, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 24000000,
    specialities: [24, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 25000000,
    specialities: [25, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 26000000,
    specialities: [26, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 27000000,
    specialities: [27, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 28000000,
    specialities: [28, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 29000000,
    specialities: [29, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 30000000,
    specialities: [30, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 31000000,
    specialities: [31, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 32000000,
    specialities: [32, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 33000000,
    specialities: [33, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 34000000,
    specialities: [34, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 35000000,
    specialities: [35, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 36000000,
    specialities: [36, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 37000000,
    specialities: [37, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 38000000,
    specialities: [38, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 39000000,
    specialities: [39, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 40000000,
    specialities: [40, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 41000000,
    specialities: [41, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 42000000,
    specialities: [42, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 43000000,
    specialities: [43, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 44000000,
    specialities: [44, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 45000000,
    specialities: [45, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 46000000,
    specialities: [46, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 47000000,
    specialities: [47, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 48000000,
    specialities: [48, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  }
]
