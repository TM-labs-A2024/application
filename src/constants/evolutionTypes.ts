export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
