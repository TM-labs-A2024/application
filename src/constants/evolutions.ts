export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
