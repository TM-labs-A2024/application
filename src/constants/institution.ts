export const institution = {
  id: '3fa85f64-5717-4562-b3fc-2c963f66afa1',
  name: 'Clínica Tex Mex',
  credentials: '14578245',
  address: 'Av. Fernandez Peña, 147.',
  email: 'test@test.com',
  phone: '+582742214441'
}
