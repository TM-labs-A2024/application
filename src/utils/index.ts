import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
