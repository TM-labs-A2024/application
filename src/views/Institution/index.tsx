import Institution from '@components/organisms/Institution'
import { Institution as InstitutionType } from '@src/types'
import React, { ReactElement } from 'react'

export default function InstitutionView({
  context
}: {
  context: {
    institution: InstitutionType
    isDenialOpen: boolean
    onDenialOpen: () => void
    onDenialClose: () => void
    isApprovalOpen: boolean
    onApprovalOpen: () => void
    onApprovalClose: () => void
    isRemovalOpen: boolean
    onRemovalOpen: () => void
    onRemovalClose: () => void
    onApproval: () => void
    onDenial: () => void
    onRemoval: () => void
    isLoading: boolean
  }
}): ReactElement {
  return <Institution context={context} />
}
