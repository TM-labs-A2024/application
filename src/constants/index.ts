import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
