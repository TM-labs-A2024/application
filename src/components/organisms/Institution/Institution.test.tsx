import { institution } from '@src/constants'
import { render, screen, fireEvent } from '@test/utils/index'

import '@testing-library/jest-dom'
import Institution from './Institution'
import 'next/navigation'

jest.mock('next/navigation')

const pendingInstitution = {
  ...institution,
  pending: true
}

describe('Organisms > Institution test', () => {
  test('The component shows the name, rif, email and phoneNumber of the institution', () => {
    render(<Institution institution={institution} />)

    const emailMatcher = (content: string) => content.includes(institution.email)
    const institutionEmail = screen.getByText(emailMatcher)

    expect(institutionEmail).toBeInTheDocument()

    const credentialsMatcher = (content: string) => content.includes(institution.credentials)
    const institutionCredentials = screen.getByText(credentialsMatcher)

    expect(institutionCredentials).toBeInTheDocument()

    expect(screen.getByText(institution.name)).toBeInTheDocument()
    expect(screen.getByText(institution.phoneNumber)).toBeInTheDocument()
  })

  test('When clicking on the access removal button, the respective confirmation modal is shown', () => {
    render(<Institution institution={institution} />)

    const deletionButton = screen.getByText('Revocar acceso')
    fireEvent.click(deletionButton)

    const confirmation = `¿Seguro que desea revocar el acceso a ${institution.name}?`

    expect(screen.getByText(confirmation)).toBeInTheDocument()
  })

  test('The component allows to remove access to an institution', () => {
    render(<Institution institution={institution} />)

    const deletionButton = screen.getByText('Revocar acceso')
    fireEvent.click(deletionButton)

    const confirmation = `¿Seguro que desea revocar el acceso a ${institution.name}?`

    expect(screen.getByText(confirmation)).toBeInTheDocument()
    expect(screen.getAllByText('Revocar acceso')).toHaveLength(2)
  })

  test("If the institution doesn't have access, when clicking on the access approval button, the respective confirmation modal is shown", () => {
    render(<Institution institution={pendingInstitution} />)

    const deletionButton = screen.getByText('Aceptar')
    fireEvent.click(deletionButton)

    const confirmation = `¿Seguro que desea conceder el acceso a ${pendingInstitution.name}?`

    expect(screen.getByText(confirmation)).toBeInTheDocument()
  })

  test('The component allows to give access to an institution', () => {
    render(<Institution institution={pendingInstitution} />)

    const deletionButton = screen.getByText('Aceptar')
    fireEvent.click(deletionButton)

    const confirmation = `¿Seguro que desea conceder el acceso a ${pendingInstitution.name}?`

    expect(screen.getByText(confirmation)).toBeInTheDocument()
    expect(screen.getByText('Conceder acceso')).toBeInTheDocument()
  })

  test("If the institution doesn't have access, when clicking on the access denial button, the respective confirmation modal is shown", () => {
    render(<Institution institution={pendingInstitution} />)

    const deletionButton = screen.getByText('Rechazar')
    fireEvent.click(deletionButton)

    const confirmation = `¿Seguro que desea denegar el acceso a ${pendingInstitution.name}?`

    expect(screen.getByText(confirmation)).toBeInTheDocument()
  })

  test('The component allows to deny access to an institution', () => {
    render(<Institution institution={pendingInstitution} />)

    const deletionButton = screen.getByText('Rechazar')
    fireEvent.click(deletionButton)

    const confirmation = `¿Seguro que desea denegar el acceso a ${pendingInstitution.name}?`

    expect(screen.getByText(confirmation)).toBeInTheDocument()
    expect(screen.getByText('Denegar acceso')).toBeInTheDocument()
  })

  test('Matches the snapshot', () => {
    render(<Institution institution={institution} />)

    expect(screen.getByTestId('institution')).toMatchSnapshot()
  })
})
