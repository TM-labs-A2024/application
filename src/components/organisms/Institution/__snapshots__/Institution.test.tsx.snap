// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > Institution test Matches the snapshot 1`] = `
<div
  class="mx-auto block h-screen w-screen overflow-hidden px-8 pb-0 pt-10 false"
  data-testid="institution"
>
  <div
    class="mb-8 flex flex-row justify-between false"
  >
    <img
      alt="logo"
      data-nimg="1"
      decoding="async"
      height="80"
      loading="lazy"
      src="/_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=640&q=75"
      srcset="/_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=256&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
  </div>
  <div
    class="mb-8 flex flex-row items-center justify-start gap-4"
  >
    <button
      aria-label="back"
      class="chakra-button css-84vx7q"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          fill="currentColor"
        />
      </svg>
    </button>
    <p
      class="chakra-text font-medium css-0"
    >
      Perfil de la institución
    </p>
  </div>
  <div
    class="flex h-4/5 flex-col justify-between lg:px-96"
  >
    <div>
      <div
        class="chakra-stack css-1uriqkk"
      >
        <h3
          class="chakra-heading css-ypohwm"
        >
          Clínica Tex Mex
        </h3>
        <p
          class="chakra-text css-0"
        >
          RIF 
          14578245
        </p>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-svjswr"
      />
      <div
        class="chakra-stack css-1j35777"
      >
        <h4
          class="text-sm text-gray-600"
        >
          Correo electrónico
        </h4>
        <p
          class="chakra-text font-medium css-0"
        >
          test@test.com
        </p>
      </div>
      <div
        class="chakra-stack css-1j35777"
      >
        <h4
          class="text-sm text-gray-600"
        >
          Teléfono
        </h4>
        <p
          class="chakra-text font-medium css-0"
        >
          +582742214441
        </p>
      </div>
    </div>
    <div
      class="mb-4 flex w-full flex-row gap-4"
    >
      <button
        class="chakra-button flex-grow css-11494p6"
        type="button"
      >
        Revocar acceso
      </button>
    </div>
  </div>
</div>
`
